### PR TITLE
Refactor/pmft tests

### PR DESCRIPTION
**Pull request by vyasr**
## Description

I consolidated most of the common testing for PMFTs into a single class.

## Motivation and Context

There was a lot of duplication in the PMFT tests that made it difficult to validate that all methods are behaving correctly without manually checking each one. This cleanup helps with that. There were also some errors with the way that bins were computed in a few places that are not actually being tested right now because most orientations in the test are being set to 0. For the most part, the underlying classes are being tested, only the methods defined in the tests themselves had problems. There are a few exceptions, for which I will make additional PRs.

## How Has This Been Tested?

Tests still pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: tests/test_pmft.py [hunk 1]
```diff
@@ -9,154 +9,153 @@
...
+        pmft = self.make_pmft()
+        self.assertEqual(str(pmft), str(eval(repr(pmft))))
+
+
+class TestPMFTR12(TestPMFT, unittest.TestCase):
+    limits = (5.23, )
+    bins = (10, 20, 30)
+    ndim = 2
+    L = 16
+    cl = freud.pmft.PMFTR12
```
Comment (bdice) [inline tests/test_pmft.py]: I'd prefer a name like `pmft_cls` over `cl`.

Comment (vyasr) [reply tests/test_pmft.py]: Done.

File: tests/test_pmft.py [hunk 2]
```diff
@@ -182,158 +181,46 @@ def test_points_ne_query_points(self):
...
+            rot_r_ij[:2] + limits) * np.asarray(self.bins[:2]) /
+            (2*limits)).astype(np.int32))
 
-    def test_repr(self):
-        max_x = 3.0
-        max_y = 4.0
-        nbins = 20
-        myPMFT = freud.pmft.PMFTXYT(max_x, max_y, nbins)
-        self.assertEqual(str(myPMFT), str(eval(repr(myPMFT))))
+        twopi = 2*np.pi
```
Comment (bdice) [inline tests/test_pmft.py]: Generally I recommend putting constants in all caps like `TWO_PI`. Perhaps even at the module level since it's reused.

Comment (vyasr) [reply tests/test_pmft.py]: Done.

File: tests/test_pmft.py [hunk 3]
```diff
@@ -9,154 +9,168 @@
...
+        correct_bin_counts[self.get_bin(
+            points[0], points[1], orientations[0], orientations[1])] = 1
+        correct_bin_counts[self.get_bin(
+            points[1], points[0], orientations[1], orientations[0])] = 1
         absoluteTolerance = 0.1
 
-        r_max = max_r
+        # No angular terms should have entries in the limits array, so this
+        # should work in all cases.
+        r_max = np.sqrt(np.sum(np.asarray(self.limits)**2))
```
Comment (bdice) [inline tests/test_pmft.py]:
```suggestion
        r_max = np.linalg.norm(self.limits)
```

File: tests/test_pmft.py [hunk 4]
```diff
@@ -755,164 +505,73 @@ def points_to_set(bin_counts):
...
-        # calculation for array idxs
-        def get_bin(query_point, point):
-            return np.floor((query_point - point + [max_x, max_y, max_z]) /
-                            [dx, dy, dz]).astype(np.int32)
-
-        bins = get_bin(points[0], points[1])
-        correct_bin_counts[bins[0], bins[1], bins[2]] = 1
-        bins = get_bin(points[1], points[0])
-        correct_bin_counts[bins[0], bins[1], bins[2]] = 1
+        """Override base class function to also test equiv orientations."""
```
Comment (bdice) [inline tests/test_pmft.py]:
```suggestion
        """Override base class function to also test equivalent orientations."""
```

File: tests/test_pmft.py [hunk 5]
```diff
@@ -755,164 +505,73 @@ def points_to_set(bin_counts):
...
+        correct_bin_counts[self.get_bin(
+            points[0], points[1], orientations[0], orientations[1])] = 1
+        correct_bin_counts[self.get_bin(
+            points[1], points[0], orientations[1], orientations[0])] = 1
         absoluteTolerance = 0.1
 
-        r_max = np.sqrt(max_x**2 + max_y**2 + max_z**2)
+        # No angular terms should have entries in the limits array, so this
+        # should work in all cases.
+        r_max = np.sqrt(np.sum(np.asarray(self.limits)**2))
```
Comment (bdice) [inline tests/test_pmft.py]:
```suggestion
        r_max = np.linalg.norm(self.limits)
```

**Review by bdice — APPROVED** [review]
Approving with minor change suggestions.

**Pull request merged**